bin/setup

```diff
@@ -13,39 +15,6 @@ set -o posix    # more strict failures in subshells
 IFS="$(printf "\n\t")"
 # ---- End unofficial bash strict mode boilerplate
 
-__dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-env_file=${__dir}/../.env
-
-function main() {
-  set -e
-
-  add_new_env_vars
-}
-
-function add_new_env_vars() {
-  # create .env and set perms if it does not exist
-  [[ ! -f "${env_file}" ]] && {
-    touch "${env_file}"
-    chmod 0600 "${env_file}"
-  }
-
-  find . -name .env.example -type f -print0 |
-    xargs -0 grep -Ehv '^\s*#' || true |
-    {
-      while IFS= read -r var; do
-        if [[ -z "${var}" ]]; then
-          continue
-        fi
-        key="${var%%=*}"        # get var key
-        var=$(eval echo "$var") # generate dynamic values
-
-        # If .env doesn't contain this env key, add it
-        if ! grep -qLE "^$key=" "${env_file}"; then
-          echo "Adding $key to .env"
-          echo "$var" >>"${env_file}"
-        fi
-      done
-    }
-}
-
-main
+find . -name .env.example -type f -print0 \
+  | xargs -0 -I '{source}' \
+    setup '{source}' .env
```

Comment:
I'm worried about `PATH` here. Should we `cd` to a known dir and then run setup with a relative path like `./bin/setup`?

Reply:
👍 👍 I didn't battle harden this very well. Just tried to open it as quickly as I could for discussion.